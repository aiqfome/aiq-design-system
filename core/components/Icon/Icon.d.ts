import React from 'react';
import { Props as FlexProps } from '../Flex';
export interface Props extends FlexProps {
    cursor?: string;
    children?: any;
    color?: string;
}
export declare const IconStyled: import("styled-components").StyledComponent<React.ForwardRefExoticComponent<FlexProps & React.RefAttributes<HTMLDivElement>>, any, Props, never>;
export declare const Icon: React.FC<Props>;
//# sourceMappingURL=Icon.d.ts.map