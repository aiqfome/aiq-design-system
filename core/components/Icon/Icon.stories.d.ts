import React from 'react';
declare const _default: {
    component: React.FC<import("./Icon").Props>;
    title: string;
    decorators: ((...args: any) => any)[];
};
export default _default;
export declare const basic: React.FC;
export declare const centralized: React.FC;
export declare const fullCentralized: React.FC;
export declare const cursor: React.FC;
//# sourceMappingURL=Icon.stories.d.ts.map