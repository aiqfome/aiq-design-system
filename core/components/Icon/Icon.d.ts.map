{"version":3,"file":"Icon.d.ts","sourceRoot":"","sources":["../../../src/components/Icon/Icon.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;AAKzB,OAAO,EAAQ,KAAK,IAAI,SAAS,EAAE,MAAM,SAAS,CAAA;AAElD,MAAM,WAAW,KAAM,SAAQ,SAAS;IACtC,MAAM,CAAC,EAAE,MAAM,CAAA;IACf,QAAQ,CAAC,EAAE,GAAG,CAAA;IACd,KAAK,CAAC,EAAE,MAAM,CAAA;CACf;AAED,eAAO,MAAM,UAAU,kJAMtB,CAAA;AAED,eAAO,MAAM,IAAI,EAAE,KAAK,CAAC,EAAE,CAAC,KAAK,CAEhC,CAAA"}