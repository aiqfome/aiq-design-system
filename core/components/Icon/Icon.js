"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Icon = exports.IconStyled = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const styled_system_1 = require("styled-system");
const Flex_1 = require("../Flex");
exports.IconStyled = styled_components_1.default(Flex_1.Flex) `
  ${styled_system_1.color}
  ${styled_system_1.space}
  ${styled_system_1.layout}

  cursor: ${props => props.cursor};
`;
exports.Icon = (_a) => {
    var { children } = _a, props = __rest(_a, ["children"]);
    return react_1.default.createElement(exports.IconStyled, Object.assign({}, props), children);
};
exports.Icon.propTypes = {
    color: prop_types_1.default.string,
    cursor: prop_types_1.default.string,
    children: prop_types_1.default.any
};
exports.Icon.defaultProps = {
    alignItems: 'center'
};
