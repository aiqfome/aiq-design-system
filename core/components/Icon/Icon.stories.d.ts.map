{"version":3,"file":"Icon.stories.d.ts","sourceRoot":"","sources":["../../../src/components/Icon/Icon.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,MAAM,OAAO,CAAA;;;;;;AAMzB,wBAIC;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAIzB,CAAA;AAED,eAAO,MAAM,WAAW,EAAE,KAAK,CAAC,EAI/B,CAAA;AAED,eAAO,MAAM,eAAe,EAAE,KAAK,CAAC,EAInC,CAAA;AAED,eAAO,MAAM,MAAM,EAAE,KAAK,CAAC,EAI1B,CAAA"}