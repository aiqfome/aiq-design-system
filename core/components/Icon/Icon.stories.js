"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.cursor = exports.fullCentralized = exports.centralized = exports.basic = void 0;
const react_1 = __importDefault(require("react"));
const md_1 = require("react-icons/md");
const addon_knobs_1 = require("@storybook/addon-knobs");
const Icon_1 = require("./Icon");
exports.default = {
    component: Icon_1.Icon,
    title: 'Icon',
    decorators: [addon_knobs_1.withKnobs]
};
exports.basic = () => (react_1.default.createElement(Icon_1.Icon, { cursor: addon_knobs_1.text('Cursor', '') },
    react_1.default.createElement(md_1.MdHome, null)));
exports.centralized = () => (react_1.default.createElement(Icon_1.Icon, { variant: 'centralized' },
    react_1.default.createElement(md_1.MdHome, null)));
exports.fullCentralized = () => (react_1.default.createElement(Icon_1.Icon, { variant: 'fullCentralized' },
    react_1.default.createElement(md_1.MdHome, null)));
exports.cursor = () => (react_1.default.createElement(Icon_1.Icon, { cursor: 'pointer' },
    react_1.default.createElement(md_1.MdHome, null)));
