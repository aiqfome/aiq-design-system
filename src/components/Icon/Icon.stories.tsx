import React from 'react'
import { MdHome } from 'react-icons/md'
import { withKnobs, text } from '@storybook/addon-knobs'

import { Icon } from './Icon'

export default {
  component: Icon,
  title: 'Icon',
  decorators: [withKnobs]
}

export const basic: React.FC = () => (
  <Icon cursor={text('Cursor', '')}>
    <MdHome />
  </Icon>
)

export const centralized: React.FC = () => (
  <Icon variant='centralized'>
    <MdHome />
  </Icon>
)

export const fullCentralized: React.FC = () => (
  <Icon variant='fullCentralized'>
    <MdHome />
  </Icon>
)

export const cursor: React.FC = () => (
  <Icon cursor='pointer'>
    <MdHome />
  </Icon>
)
