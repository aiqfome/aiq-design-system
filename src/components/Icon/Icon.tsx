import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { color, space, layout } from 'styled-system'

import { Flex, Props as FlexProps } from '../Flex'

export interface Props extends FlexProps {
  cursor?: string
  children?: any
  color?: string
}

export const IconStyled = styled(Flex)<Props>`
  ${color}
  ${space}
  ${layout}

  cursor: ${props => props.cursor};

  align-items: center;
`

export const Icon: React.FC<Props> = ({ children, ...props }) => {
  return <IconStyled {...props}>{children}</IconStyled>
}

Icon.propTypes = {
  color: PropTypes.string,
  cursor: PropTypes.string,
  children: PropTypes.any
}
