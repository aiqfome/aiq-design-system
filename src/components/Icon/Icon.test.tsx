import React from 'react'

import { MdStore } from 'react-icons/md'
import { render } from '../utils/test/render'

import { Icon } from '../Icon'

describe('Icon', () => {
  it('should have to render without crashing', () => {
    const component = render(
      <Icon color='primary'>
        <MdStore />
      </Icon>
    )
    expect(component).toBeTruthy()
  })
})
