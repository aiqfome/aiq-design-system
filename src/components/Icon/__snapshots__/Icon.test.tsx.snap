// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<div
  className="sc-bdnylx fSUZen sc-gtssRu hgFtFi"
>
  <span
    className="sc-dlnjPT ljXqAF"
  >
    Icon component!
  </span>
</div>
`;
